Sources/FoundationEssentials/FileManager/FileManager+Directories.swift

```diff
@@ -466,13 +466,20 @@ extension _FileManagerImpl {
     
     var currentDirectoryPath: String? {
 #if os(Windows)
-        let dwLength: DWORD = GetCurrentDirectoryW(0, nil)
-        return withUnsafeTemporaryAllocation(of: WCHAR.self, capacity: Int(dwLength)) {
-            if GetCurrentDirectoryW(dwLength, $0.baseAddress) == dwLength - 1 {
-                return String(decodingCString: $0.baseAddress!, as: UTF16.self)
+        for _ in 0 ... 8 {
+            let dwLength: DWORD = GetCurrentDirectoryW(0, nil)
+            guard dwLength > 0 else { return nil }
```

Comment:
Do you know when `dwLength` can be 0 here? Just because I’m assuming that `currentDirectoryPath` is not `nil` in https://github.com/swiftlang/swift-foundation/pull/1033.

Reply:
No, I don't know _when_ that can be 0, but the docs indicate that it is possible for the call to fail and in such a case `0` will be returned.

https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-getcurrentdirectory

> If the function fails, the return value is zero. To get extended error information, call [GetLastError](https://learn.microsoft.com/en-us/windows/desktop/api/errhandlingapi/nf-errhandlingapi-getlasterror).

At the same time, it does document that each process has a working directory, so it is certainly unclear when and how this function could fail, especially when simply querying the required buffer size.

Reply:
Hmm, OK. I guess returning `nil` is the best option.